src/dbt_client/dbtClient.ts

```diff
@@ -0,0 +1,124 @@
+import { OutputChannel, window } from 'vscode';
+import { OnSourceFileChanged, SourceFileChangedEvent } from '../manifest/sourceFileChangedEvent';
+import { CommandProcessExecution } from '../utils';
+import { dbtClientCommandQueue as dbtClientCommandQueue } from './dbtClientCommandQueue';
+
+enum DBT {
+	Install,
+	Update
+}
+
+enum PromptAnswer {
+	Yes = "Yes",
+	No = "No"
+}
+
+class DBTClient implements OnSourceFileChanged {
+	public pythonPath!: string;
```

Comment:
I would say the pythonPath is a required constructor parameter for the DBTClient, adding the exclamation mark WILL generate runtime exceptions. as pythonPath can be null if you don't select a python interpreter.